### DISCUSSION TITLE
Normal mode for picker batch operations

### DISCUSSION THREAD

**Discussion by QiBaobin** [helix-editor/helix#10600] [Ideas]
#### Describe your feature request

It would be make it very powerful if we can go to a normal mode by clicking Esc in Space mode, which would let us operate all files of current picker showing.  The preview panel can show only selected file's motion. 

This would definitely a good replacement of vim's `bufdo`, `cfdo` etc.

**Comment by pickfire** [comment]
This reminds me of doom emacs search and replace across the whole project, copy the global search picker into a buffer and let you edit directly, and when you save it will modify the content. It also have the same concept where you can just modify the file name as well.

**Comment by axyz** [comment]
partially related: is there any other suggested way currently available to do project wide search/replace? I'm thinking also at how to replace the typical workflow of sending stuff to quickfix (e.g. via :grep configured to use ripgrep) and then do some `:cdo` / `:cfdo` commands

**Comment by sudormrfbin** [comment]
@axyz LSP has a rename symbol function (bound to `space + r`) which could be used in lieu of search and replace for most cases but other than that there is no dedicated search and replace interface. `space + /` does project wide search, and `space + '` can be used to bring back the same search results again after dismissing the global search picker (also we don't have the quickfix lists yet).
